### ISSUE TITLE
[NuGet.org Bug]: "This package has not been indexed yet." for more than 3 hours

### ISSUE THREAD

**Issue by Piedone** [NuGet/NuGetGallery#8945]
### Impact

It's more difficult to complete my work

### Describe the bug

Similar to [the other such issues here](https://github.com/NuGet/NuGetGallery/issues?q=is%3Aissue+This+package+has+not+been+indexed+yet.+is%3Aclosed) or [in the Home repo](https://github.com/NuGet/Home/issues?q=is%3Aissue+This+package+has+not+been+indexed+yet.+is%3Aclosed), our freshly released [Lombiq.HelpfulLibraries package](https://www.nuget.org/packages/Lombiq.HelpfulLibraries/) is still being indexed, 3 hours after publishing it. This doesn't seem to be normal.

### Repro Steps

1. I published the package, the first version for this package, with `dotnet nuget push` from [this GitHub Action](https://github.com/Lombiq/Helpful-Libraries/actions/runs/1715042166).
2. The https://www.nuget.org/packages/Lombiq.HelpfulLibraries page appeared within seconds, as expected.
3. Initially the "This package has not been indexed yet." message was displayed, as expected, however, this still is there. I'd expect the latest version(s) (the ones I've published since then) to gradually get indexed but this is for the first version:
![image](https://user-images.githubusercontent.com/1976647/150044061-83e84369-75b6-4569-913d-7c1a48d49df9.png)

This is the case for all other packages I've published at the same time:

- [Lombiq.HelpfulLibraries.LinqToDb](https://www.nuget.org/packages/Lombiq.HelpfulLibraries.LinqToDb/0.0.4)
- [Lombiq.HelpfulLibraries.RestEase](https://www.nuget.org/packages/Lombiq.HelpfulLibraries.RestEase/0.0.4)
- [Lombiq.HelpfulLibraries.Targets](https://www.nuget.org/packages/Lombiq.HelpfulLibraries.Targets/0.0.4)

### Expected Behavior

The package is indexed after a couple of minutes, half an hour at most. If there are any temporary issues then I'd expect https://status.nuget.org/ to show it but it's all green:
![image](https://user-images.githubusercontent.com/1976647/150043880-99f27937-cfa8-434e-8871-468d11862f53.png)

### Screenshots

Added above.

**Comment by joelverhagen** [comment]
Hello @Piedone, apologies on the delay for indexing your packages. NuGet.org got behind on the queue for new packages. You're right, this is pretty irregular, and our team is investigating the root cause.

We should be completely caught up now. It looks like your packages are fully indexed. Please let us know if you encounter this problem again.

@dannyjdev - could you include post-mortem an investigation or action item about why our monitoring system didn't automatically update the status page?

**Comment by Piedone** [comment]
Thank you. It would also be great to see some hints on the expected indexing time in the message, together with information on what you can do (or where you can get support) if indexing is not done withing that time frame.

**Comment by joelverhagen** [comment]
@clairernovotny - what are your thoughts on improving messaging here on NuGet.org? Maybe something like a static message about 95th percentile ingestion times (we have this data) and a link to the contact support form?

**Comment by Piedone** [comment]
I published a new package and briefly got a message (couldn't screenshot it) pointing to this: https://docs.microsoft.com/en-us/nuget/nuget-org/publish-a-package#package-validation-and-indexing During this, the status of the package was "Validating" and now it's "Listed", and I get the same indexing message as before. This validation warning is actually exactly what I'd like to see when just indexing remains too: informative, tells about usual and maximal times, points to support resources and the status page.

**Comment by Piedone** [comment]
This is also an example of a helpful warning:
![image](https://user-images.githubusercontent.com/1976647/150584407-ae2779f6-5235-4c69-92de-2f4a0bf8cc20.png)

My first though was "why is my icon now showing?" and it answered it precisely.

**Comment by joelverhagen** [comment]
Thanks for the feedback @Piedone. I spoke to @clairernovotny (the PM who owns this experience) and here are some thoughts:

1. We can add a configuration for NuGet.org which enhances this message to "Indexing is taking longer than usual. Try checking the [status page](https://status.nuget.org/) or reaching out to support@nuget.org." when the time since package creation exceeds some threshold.
2. Fix our bug on the status page which prevented it from going "yellow" when the lag exceeds this same threshold.

In other words, we have two issues to fix here! Thanks for your feedback!

@dannyjdev, let's include both of these things in the Post-Mortem action items.

**Comment by mhazamidev** [comment]
https://www.nuget.org/packages/Tafi/8.5.14
This issue still occurs, Mine is taking up to 4 hours now.

**Comment by HughPH** [comment]
Mine was still not indexed for about 14 hours. Which is OK because I messed it up after not enough sleep! But I pushed again today and the issue persists. I'll keep watching this.

**Comment by joelverhagen** [comment]
Hey folks, we are actively working on this issue. Some corrupt package metadata made its way into our publishing pipeline, causing downstream jobs to get blocked. We are performing some manual data fixes to mitigate the issue and will follow up with a bug fix after that.

We apologies for the delay and will get your packages published as soon as possible!

**Comment by joelverhagen** [comment]
Our manual data fixes are complete. We are seeing our publishing jobs catch up. I will close this issue when we see all delayed packages ingested and available in our search index. We will track the root cause fix separately.

**Comment by HughPH** [comment]
Many thanks for the updates

**Comment by joelverhagen** [comment]
Recent packages are now flowing as expected. We've fully caught up. Please open a new issue if you see any further problems.
https://www.nuget.org/packages?q=&includeComputedFrameworks=true&prerel=true&sortby=created-desc

<img width="899" alt="Image" src="https://github.com/user-attachments/assets/4ba36eb3-7dd6-4994-a79d-9a79924734e5" />

**Comment by lyndaidaii** [comment]
Thank you for patience. The issue has been mitigated.

**Comment by Afreed-cp** [comment]
This issue still occurs, Mine is taking more than 2 hours now.
https://www.nuget.org/packages/UmbracoGenie/

**Comment by lyndaidaii** [comment]
Hey folks, we're actively working on this issue. Some corrupt package metadata made its way into our publishing pipeline, which is currently blocking downstream jobs. We're performing manual data fixes to mitigate the issue, and we are working on the short-term solution. 
We apologize for the delay and appreciate your patience — we'll get your packages published as soon as possible!

**Comment by Alerinos** [comment]
![Image](https://github.com/user-attachments/assets/ab3ce57a-8c7d-41f9-be41-1aa81dc4849b)

Same problem, I have been waiting for 2 hours. I even bumped the package thinking something was hung up.

**Comment by mhazamidev** [comment]
I've got the same problem 
This package has not been indexed yet. It will appear in search results and will be available for install/restore after indexing is complete.

https://www.nuget.org/packages/CPC

**Comment by Sicos1977** [comment]
Same issue today
